Marlin/src/Marlin.cpp

```diff
@@ -218,15 +218,40 @@ void setup_killpin() {
   #endif
 }
 
-#if ENABLED(FILAMENT_RUNOUT_SENSOR)
+if ENABLED(FILAMENT_RUNOUT_MULTI_SENSORS)
 
-  void setup_filrunoutpin() {
+  void setup_filrunoutpin() { 
     #if ENABLED(ENDSTOPPULLUP_FIL_RUNOUT)
-      SET_INPUT_PULLUP(FIL_RUNOUT_PIN);
+        SET_INPUT_PULLUP(FIL_RUNOUT_0_PIN);
+	  #if FIL_RUNOUT_1_PIN > -1
+	    SET_INPUT_PULLUP(FIL_RUNOUT_1_PIN);
+	  #endif
+	  #if FIL_RUNOUT_2_PIN > -1
+	    SET_INPUT_PULLUP(FIL_RUNOUT_2_PIN);  
+      #endif
+      #if FIL_RUNOUT_3_PIN > -1
+	    SET_INPUT_PULLUP(FIL_RUNOUT_3_PIN);     
+	  #endif
+	  #if FIL_RUNOUT_4_PIN > -1
+	    SET_INPUT_PULLUP(FIL_RUNOUT_4_PIN);     
+      #endif
+	
     #else
-      SET_INPUT(FIL_RUNOUT_PIN);
+        SET_INPUT(FIL_RUNOUT_0_PIN);
+      #if FIL_RUNOUT_2_PIN > -1
+	    SET_INPUT(FIL_RUNOUT_1_PIN);
+	  #endif
+	  #if FIL_RUNOUT_3_PIN > -1
+	    SET_INPUT(FIL_RUNOUT_2_PIN);  
+      #endif
+      #if FIL_RUNOUT_4_PIN > -1
+	    SET_INPUT(FIL_RUNOUT_3_PIN);     
+	  #endif
+	  #if FIL_RUNOUT_PIN_5 > -1
+	    SET_INPUT(FIL_RUNOUT_4_PIN);     
     #endif
```

Comment:
Sensor Reading  ! in multi sensor config  only the sensor of the active extruder is read , and it's PERFECT